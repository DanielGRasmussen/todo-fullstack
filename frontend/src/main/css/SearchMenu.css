/* Menu */
#menu {
	display: grid;
	grid-template-columns: 1fr 1fr;
	grid-template-rows: 30px 60px 60px 30px 60px;
	grid-gap: 15px;
	margin-bottom: 25px;
}

#filters-label {
	grid-column: 1 / 3;
	grid-row: 1 / 2;
	margin-top: 16px;
	text-align: center;
	cursor: default;
}

#sorts-label {
	grid-column: 1 / 3;
	grid-row: 4 / 5;
	margin-top: 16px;
	text-align: center;
	cursor: default;
}

/* Filters */
#status {
	grid-column: 1 / 2;
	grid-row: 2 / 3;
	margin-left: auto;
}

#type {
	grid-column: 1 / 2;
	grid-row: 3 / 4;
	margin-left: auto;
}

/* Datepicker */
#datepicker-wrapper {
	width: 372px;
	grid-column: 2 / 3;
	grid-row: 2 / 3;
}

#datepicker {
	width: 359px;
	height: 46px;
	font-size: 12pt;
	border-radius: 5px;
	border: 1px solid #9e9e9e;
	padding-left: 7px;

	grid-column: 2 / 3;
	display: block;
}

#datepicker:focus {
	border: 1px solid #2684ff;
}

#time {
	grid-column: 2 / 3;
	grid-row: 3 / 4;
}

/* Selects sorts */
#menu #sorts {
	display: flex;
	grid-column: 1 / 3;
	grid-row: 5 / 6;
	margin: 0 auto;
}

/* Search bar */
#search-bar {
	width: 250px;
	height: 46px;
	font-size: 12pt;
	border: 1px solid #9e9e9e;
	border-radius: 4px;
	padding-left: 8px;
}

@media screen and (min-width: 801px) {
	#search-bar-container-container {
		width: 100vw;
		position: fixed;
		top: -100px;
		left: 0;
		z-index: 1;
		transition: top 0.3s ease-in-out;
		height: 70px;
	}

	#search-bar-container {
		width: 80%;
		position: relative;
		top: 0;
		min-width: 800px;
		max-width: 1100px;
		margin: 0 auto;
	}

	#menu #search-bar {
		position: absolute;
		top: 117px;
		right: 15px;
		height: 40px;
		border-radius: 15px;
	}

	#menu #search-bar-container-container.hide {
		/* This is -200px instead of -100px so the animation speed stays the same and the containers don't block the nav */
		top: -200px;
	}
}

/* Tablet/mobile SearchMenu */
@media screen and (max-width: 800px) {
	#menu {
		display: block;
	}

	#menu label {
		margin-top: 10px;
		display: block;
	}

	#search-bar {
		width: 80%;
		margin: 0 auto;
		display: block;
	}

	#menu > div {
		margin: 0 auto;
		width: 80%;
		min-width: 300px;
	}

	#datepicker-wrapper {
		width: 100%;
	}

	#datepicker {
		width: calc(100% - 11px);
	}

	.css-b62m3t-container .css-u5xlfi-control,
	.css-b62m3t-container .css-10v2721-control {
		width: 100%;
		margin: 0 auto;
	}

	#sort-select {
		width: calc(100% - 50px);
	}

	.dropdown-item-header {
		cursor: pointer;
		position: relative;
	}

	.arrow {
		position: absolute;
		top: 13px;
		right: 20px;
		width: 0;
		height: 0;
		margin-left: 5px;
		border-style: solid;
		border-width: 5px 4px 0 4px;
		border-color: #4e4e4e transparent transparent transparent;
		transition: transform 0.3s ease-in-out;
	}

	.arrow.up {
		transform: rotate(180deg);
	}

	.dropdown-item-content {
		transition: all 0.3s ease-in-out;
		height: 0;
		overflow: hidden;
	}

	/* fit-content doesn't show the animation for whatever reason */
	.dropdown-item-content.filters.show {
		height: 324px;
	}

	.dropdown-item-content.sorts.show {
		height: 50px;
	}
}

/* React-select animations */
.select-menu {
	animation: fadeIn 0.2s ease-in-out;
}

.select-menu-close {
	animation: fadeOut 0.3s ease-in-out;
}

@keyframes fadeIn {
	from {
		opacity: 0;
		transform: translateY(2rem);
	}
	to {
		opacity: 1;
		transform: translateY(0);
	}
}

@keyframes fadeOut {
	from {
		opacity: 1;
		transform: translateY(0);
	}
	to {
		opacity: 0;
		transform: translateY(2rem);
	}
}

/* Style for the tooltips */
.tooltip {
	position: relative;
}

.tooltip::before {
	display: none;
	position: absolute;
	background-color: black;
	color: white;
	padding: 5px;
	border-radius: 5px;
	top: -30px;
	left: 50%;
	transform: translateX(-50%);
	text-align: center;
}

/* Show the tooltip on hover */
.tooltip:hover::before {
	display: block;
}

/* Sort-direction title message in SortButton.css */
