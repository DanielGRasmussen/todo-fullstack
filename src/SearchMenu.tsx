import React, { useState } from "react";
import Select, { components } from "react-select";

function SortButton({ onClick }): JSX.Element {
	const [isDescending, setIsDescending] = useState(true);
	const [first, isFirst] = useState(true);

	// down arrow = descending = z to a
	const handleOnClick = (): void => {
		setIsDescending(!isDescending);
		isFirst(false);
		onClick(isDescending);
	};

	let sortBy: string;
	if (isDescending) {
		sortBy = "Ascending";
	} else {
		sortBy = "Descending";
	}

	// Swap between rotating it upside-down and downside-up
	const arrowStyle: { animation: string } = {
		animation: isDescending
			? "downside-up 0.3s linear forwards"
			: "upside-down 0.3s linear forwards"
	};
	// Since it requires the same type.
	const blankStyle: { animation: string } = { animation: "" };

	return (
		<div onClick={handleOnClick} id="sort-direction">
			{isDescending ? (
				<img
					src={
						process.env.PUBLIC_URL + first
							? "/assets/up_arrow.svg"
							: "/assets/down_arrow.svg"
					}
					alt="Ascending order"
					style={first ? blankStyle : arrowStyle}
					title={sortBy}
				/>
			) : (
				<img
					src={process.env.PUBLIC_URL + "/assets/up_arrow.svg"}
					alt="Descending order"
					style={arrowStyle}
					title={sortBy}
				/>
			)}
		</div>
	);
}

function SearchMenu(
	searchQuery: string,
	setSearchQuery,
	sortingOptions: { value: string; label: string }[],
	selectedSortingOption: { value: string; label: string },
	setSelectedSortingOption,
	handleSortOrderChange
): JSX.Element {
	const [uniqueId] = React.useState(
		() => "select_" + Math.random().toFixed(5).slice(2)
	);

	const selectStyles = {
		control: (provided, state) => ({
			...provided,
			background: "#fff",
			borderColor: "#9e9e9e",
			minHeight: "50px",
			height: "50px",
			boxShadow: state.isFocused ? null : null,
			width: "370px",
			cursor: "pointer"
		}),
		valueContainer: (provided) => ({
			...provided,
			height: "30px",
			padding: "0 6px"
		}),
		input: (provided) => ({
			...provided
		}),
		indicatorSeparator: () => ({
			display: "none"
		}),
		indicatorsContainer: (provided) => ({
			...provided,
			height: "40px",
			marginTop: "5px"
		}),
		menu: (provided) => ({
			...provided,
			marginTop: "-5px",
			width: "370px"
		})
	};

	return (
		<form id="menu">
			<input
				id="search-bar"
				type="text"
				value={searchQuery}
				onChange={(event) => setSearchQuery(event.target.value)}
				placeholder="Search..."
			/>
			<Select
				id={uniqueId}
				options={sortingOptions}
				value={selectedSortingOption}
				onChange={(option) => setSelectedSortingOption(option)}
				styles={selectStyles}
				components={{
					Menu: (props) => (
						<components.Menu
							children=""
							{...props}
							className="select-menu"
						/>
					)
				}}
				onMenuClose={() => {
					// Can't just do event handlers with class toggle and timeout
					// Menu reappears for a split second and animation goes ultra-fast if you do.
					const menu = document.querySelector(
						`#${uniqueId} .select-menu`
					);
					const container: HTMLElement = menu?.parentElement;
					const clonedMenu = menu?.cloneNode(true) as HTMLElement;

					if (!clonedMenu) return;

					clonedMenu.classList.add("select-menu-close");
					clonedMenu.addEventListener("animationend", () => {
						container?.removeChild(clonedMenu);
					});

					container?.appendChild(clonedMenu);
				}}
			/>
			<SortButton onClick={handleSortOrderChange} />
		</form>
	);
}

export default SearchMenu;
