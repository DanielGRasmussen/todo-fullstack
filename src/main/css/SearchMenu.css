/* Menu */
#menu {
	display: flex;
	justify-content: center;
	flex-direction: column;
	flex-wrap: wrap;
	align-items: center;
}

/* Search bar */
#search-bar {
	width: 250px;
	height: 46px;
	font-size: 12pt;
	border: 1px solid #9e9e9e;
	border-radius: 4px;
	padding-left: 8px;
}

@media screen and (min-width: 800px) {
	#search-bar-container-container {
		width: 100vw;
		position: fixed;
		top: -100px;
		left: 0;
		z-index: 1;
		transition: top 0.3s ease-in-out;
		height: 70px;
	}

	#search-bar-container {
		width: 80%;
		position: relative;
		top: 0;
		min-width: 800px;
		max-width: 1100px;
		margin: 0 auto;
	}

	#menu #search-bar {
		position: absolute;
		top: 117px;
		right: 15px;
		height: 40px;
		border-radius: 15px;
	}

	#menu #search-bar-container-container.hide {
		/* This is -200px instead of -100px so the animation speed stays the same and the containers don't block the nav */
		top: -200px;
	}
}

/* React Select */
.css-b62m3t-container {
	margin-top: 10px;
}

@media screen and (max-width: 500px) {
	/* The class name of the react-select control changes when the dropdown menu is up */
	.css-b62m3t-container {
		width: 245px;
		margin-left: 10px;
		padding: 0;
	}

	.css-b62m3t-container > div {
		width: 225px;
	}
}

#timeframe {
	width: 180px;
	margin: 0 0 0 25px;
}

#timeframe > div {
	width: 160px;
}

/* React-select animations */
@keyframes fadeIn {
	from {
		opacity: 0;
		transform: translateY(2rem);
	}
	to {
		opacity: 1;
		transform: translateY(0);
	}
}

@keyframes fadeOut {
	from {
		opacity: 1;
		transform: translateY(0);
	}
	to {
		opacity: 0;
		transform: translateY(2rem);
	}
}

.select-menu {
	animation: fadeIn 0.2s ease-in-out;
}

.select-menu-close {
	animation: fadeOut 0.3s ease-in-out;
}

/* Date Picker */
#dates {
	display: block;
	margin-top: 10px;
}

@media screen and (min-width: 500px) {
	#dates {
		display: flex;
	}
}

#datepicker-wrapper {
	width: 213px;
}

#datepicker {
	width: 200px;
	height: 46px;
	font-size: 12pt;
	border-radius: 5px;
	border: 1px solid #9e9e9e;
	padding-left: 7px;
}

/* Selects organization */
#menu #sorts {
	display: flex;
}

/* Style for the tooltips */
.tooltip {
	position: relative;
}

.tooltip::before {
	display: none;
	position: absolute;
	background-color: black;
	color: white;
	padding: 5px;
	border-radius: 5px;
	top: -30px;
	left: 50%;
	transform: translateX(-50%);
	text-align: center;
}

/* Show the tooltip on hover */
.tooltip:hover::before {
	display: block;
}

/* Sort-direction title message in SortButton.css */
