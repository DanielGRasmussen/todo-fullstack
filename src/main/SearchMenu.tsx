import "./css/SearchMenu.css";
import React from "react";
import makeAnimated from "react-select/animated";
import Select, { components } from "react-select";
import SortButton from "./SortButton";

const animatedComponents = makeAnimated();
const customComponents = {
	Menu: (props) => (
		<components.Menu children="" {...props} className="select-menu" />
	)
};

function SearchMenu(
	searchQuery: string,
	setSearchQuery,
	sortingOptions: { value: string; label: string }[],
	selectedSortingOption: { value: string; label: string }[],
	setSelectedSortingOption,
	handleSortOrderChange
): JSX.Element {
	const [uniqueId] = React.useState(
		() => "select_" + Math.random().toFixed(5).slice(2)
	);

	// Makes react-select thinner
	const selectStyles = {
		control: (provided) => ({
			...provided,
			background: "#fff",
			borderColor: "#9e9e9e",
			minHeight: "50px",
			height: "50px",
			boxShadow: null,
			width: "370px",
			cursor: "pointer"
		}),
		valueContainer: (provided) => ({
			...provided,
			height: "30px",
			padding: "0 6px"
		}),
		indicatorSeparator: () => ({
			display: "none"
		}),
		indicatorsContainer: (provided) => ({
			...provided,
			height: "40px",
			marginTop: "5px"
		}),
		menu: (provided) => ({
			...provided,
			marginTop: "-5px",
			width: "370px"
		})
	};

	return (
		<form id="menu" autoComplete="off">
			<div id="search-bar-container-container">
				{/* First container for absolute positioning and 100% width */}
				<section id="search-bar-container">
					{/*
					 * Second container to be the same width as the header/main then use margins to move it. (Fancy CSS could probably replace that)
					 * Section instead of div because my IDE and ESLint didn't agree on how it should be formatted.
					 * This was not passed into the header function because the functions this calls on didn't to update the screen.
					 */}
					<input
						id="search-bar"
						type="text"
						value={searchQuery}
						onChange={(event) => setSearchQuery(event.target.value)}
						placeholder="Search..."
					/>
				</section>
			</div>
			<Select
				id={uniqueId}
				options={sortingOptions}
				value={selectedSortingOption}
				onChange={(option) => setSelectedSortingOption(option)}
				styles={selectStyles}
				isMulti
				components={{ ...animatedComponents, ...customComponents }}
				onMenuClose={() => {
					// Can't just do event handlers with class toggle and timeout
					// Menu reappears for a split second and animation goes ultra-fast if you do.
					const menu = document.querySelector(
						`#${uniqueId} .select-menu`
					);
					const container: HTMLElement = menu?.parentElement;
					const clonedMenu = menu?.cloneNode(true) as HTMLElement;

					if (!clonedMenu) return;

					clonedMenu.classList.add("select-menu-close");
					clonedMenu.addEventListener("animationend", () => {
						container?.removeChild(clonedMenu);
					});

					container?.appendChild(clonedMenu);
				}}
			/>
			<SortButton onClick={handleSortOrderChange} />
		</form>
	);
}

export default SearchMenu;
