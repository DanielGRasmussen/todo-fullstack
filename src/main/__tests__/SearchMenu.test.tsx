import React from "react";
import { render } from "@testing-library/react";
import SearchMenu from "../SearchMenu";

describe("SearchMenu", () => {
	const setSearchQuery = jest.fn();
	const setSelectedSortingOption = jest.fn();
	const setStatusFilter = jest.fn();
	const setSortOrder = jest.fn();
	const setFilters = jest.fn();

	const props = {
		searchQuery: "Test Search Query",
		setSearchQuery,
		sortingOptions: [
			{ value: "option1", label: "Option 1" },
			{ value: "option2", label: "Option 2" }
		],
		selectedSortingOption: [{ value: "option1", label: "Option 1" }],
		setSelectedSortingOption,
		sortOrder: true,
		setSortOrder,
		currentStatusFilters: [{ value: "incomplete", label: "Incomplete" }],
		setStatusFilter,
		filterOptions: [
			{ value: "filter1", label: "Filter 1" },
			{ value: "filter2", label: "Filter 2" }
		],
		filters: [],
		setFilters
	};

	it("should render the component", () => {
		const { container } = render(<SearchMenu {...props} />);
		expect(container).toMatchSnapshot();
	});
});
