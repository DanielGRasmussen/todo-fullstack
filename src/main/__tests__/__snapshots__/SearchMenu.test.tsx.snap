// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchMenu should render the component 1`] = `
<div>
  <form
    autocomplete="off"
    id="menu"
  >
    <div
      id="search-bar-container-container"
    >
      <section
        id="search-bar-container"
      >
        <input
          id="search-bar"
          placeholder="Search..."
          type="text"
          value=""
        />
      </section>
    </div>
    <div
      class=" css-b62m3t-container"
      id="status-filter"
    >
      <span
        class="css-1f43avz-a11yText-A11yText"
        id="react-select-2-live-region"
      />
      <span
        aria-atomic="false"
        aria-live="polite"
        aria-relevant="additions text"
        class="css-1f43avz-a11yText-A11yText"
      />
      <div
        class=" css-u5xlfi-control"
      >
        <div
          class=" css-525zur-ValueContainer"
          style="display: flex;"
        >
          <div
            style="overflow: hidden; white-space: nowrap; width: auto;"
          >
            <div
              class=" css-1p3m7a8-multiValue"
            >
              <div
                class=" css-wsp0cs-MultiValueGeneric"
              >
                Incomplete
              </div>
              <div
                aria-label="Remove Incomplete"
                class=" css-12a83d4-MultiValueRemove"
                role="button"
              >
                <svg
                  aria-hidden="true"
                  class="css-tj5bde-Svg"
                  focusable="false"
                  height="14"
                  viewBox="0 0 20 20"
                  width="14"
                >
                  <path
                    d="M14.348 14.849c-0.469 0.469-1.229 0.469-1.697 0l-2.651-3.030-2.651 3.029c-0.469 0.469-1.229 0.469-1.697 0-0.469-0.469-0.469-1.229 0-1.697l2.758-3.15-2.759-3.152c-0.469-0.469-0.469-1.228 0-1.697s1.228-0.469 1.697 0l2.652 3.031 2.651-3.031c0.469-0.469 1.228-0.469 1.697 0s0.469 1.229 0 1.697l-2.758 3.152 2.758 3.15c0.469 0.469 0.469 1.229 0 1.698z"
                  />
                </svg>
              </div>
            </div>
          </div>
          <div
            class=" css-qbdosj-Input"
            data-value=""
          >
            <input
              aria-autocomplete="list"
              aria-expanded="false"
              aria-haspopup="true"
              autocapitalize="none"
              autocomplete="off"
              autocorrect="off"
              class=""
              id="react-select-2-input"
              role="combobox"
              spellcheck="false"
              style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
              tabindex="0"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class=" css-1gscolb-IndicatorsContainer"
        >
          <div
            aria-hidden="true"
            class=" css-1xc3v61-indicatorContainer"
          >
            <svg
              aria-hidden="true"
              class="css-tj5bde-Svg"
              focusable="false"
              height="20"
              viewBox="0 0 20 20"
              width="20"
            >
              <path
                d="M14.348 14.849c-0.469 0.469-1.229 0.469-1.697 0l-2.651-3.030-2.651 3.029c-0.469 0.469-1.229 0.469-1.697 0-0.469-0.469-0.469-1.229 0-1.697l2.758-3.15-2.759-3.152c-0.469-0.469-0.469-1.228 0-1.697s1.228-0.469 1.697 0l2.652 3.031 2.651-3.031c0.469-0.469 1.228-0.469 1.697 0s0.469 1.229 0 1.697l-2.758 3.152 2.758 3.15c0.469 0.469 0.469 1.229 0 1.698z"
              />
            </svg>
          </div>
          <span
            class=" css-1xjgjl1-IndicatorSeparator"
          />
          <div
            aria-hidden="true"
            class=" css-1xc3v61-indicatorContainer"
          >
            <svg
              aria-hidden="true"
              class="css-tj5bde-Svg"
              focusable="false"
              height="20"
              viewBox="0 0 20 20"
              width="20"
            >
              <path
                d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
    <div
      class=" css-b62m3t-container"
      id="filter-select"
    >
      <span
        class="css-1f43avz-a11yText-A11yText"
        id="react-select-3-live-region"
      />
      <span
        aria-atomic="false"
        aria-live="polite"
        aria-relevant="additions text"
        class="css-1f43avz-a11yText-A11yText"
      />
      <div
        class=" css-u5xlfi-control"
      >
        <div
          class=" css-qsn7pb-ValueContainer"
          style="display: grid;"
        >
          <div
            class=" css-1jqq78o-placeholder"
            id="react-select-3-placeholder"
          >
            Type filter
          </div>
          <div
            class=" css-qbdosj-Input"
            data-value=""
          >
            <input
              aria-autocomplete="list"
              aria-describedby="react-select-3-placeholder"
              aria-expanded="false"
              aria-haspopup="true"
              autocapitalize="none"
              autocomplete="off"
              autocorrect="off"
              class=""
              id="react-select-3-input"
              role="combobox"
              spellcheck="false"
              style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
              tabindex="0"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class=" css-1gscolb-IndicatorsContainer"
        >
          <span
            class=" css-1xjgjl1-IndicatorSeparator"
          />
          <div
            aria-hidden="true"
            class=" css-1xc3v61-indicatorContainer"
          >
            <svg
              aria-hidden="true"
              class="css-tj5bde-Svg"
              focusable="false"
              height="20"
              viewBox="0 0 20 20"
              width="20"
            >
              <path
                d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
    <div
      id="sorts"
    >
      <div
        class=" css-b62m3t-container"
        id="sort-select"
      >
        <span
          class="css-1f43avz-a11yText-A11yText"
          id="react-select-4-live-region"
        />
        <span
          aria-atomic="false"
          aria-live="polite"
          aria-relevant="additions text"
          class="css-1f43avz-a11yText-A11yText"
        />
        <div
          class=" css-u5xlfi-control"
        >
          <div
            class=" css-525zur-ValueContainer"
            style="display: flex;"
          >
            <div
              style="overflow: hidden; white-space: nowrap; width: auto;"
            >
              <div
                class=" css-1p3m7a8-multiValue"
              >
                <div
                  class=" css-wsp0cs-MultiValueGeneric"
                >
                  Option 1
                </div>
                <div
                  aria-label="Remove Option 1"
                  class=" css-12a83d4-MultiValueRemove"
                  role="button"
                >
                  <svg
                    aria-hidden="true"
                    class="css-tj5bde-Svg"
                    focusable="false"
                    height="14"
                    viewBox="0 0 20 20"
                    width="14"
                  >
                    <path
                      d="M14.348 14.849c-0.469 0.469-1.229 0.469-1.697 0l-2.651-3.030-2.651 3.029c-0.469 0.469-1.229 0.469-1.697 0-0.469-0.469-0.469-1.229 0-1.697l2.758-3.15-2.759-3.152c-0.469-0.469-0.469-1.228 0-1.697s1.228-0.469 1.697 0l2.652 3.031 2.651-3.031c0.469-0.469 1.228-0.469 1.697 0s0.469 1.229 0 1.697l-2.758 3.152 2.758 3.15c0.469 0.469 0.469 1.229 0 1.698z"
                    />
                  </svg>
                </div>
              </div>
            </div>
            <div
              class=" css-qbdosj-Input"
              data-value=""
            >
              <input
                aria-autocomplete="list"
                aria-expanded="false"
                aria-haspopup="true"
                autocapitalize="none"
                autocomplete="off"
                autocorrect="off"
                class=""
                id="react-select-4-input"
                role="combobox"
                spellcheck="false"
                style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                tabindex="0"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class=" css-1gscolb-IndicatorsContainer"
          >
            <div
              aria-hidden="true"
              class=" css-1xc3v61-indicatorContainer"
            >
              <svg
                aria-hidden="true"
                class="css-tj5bde-Svg"
                focusable="false"
                height="20"
                viewBox="0 0 20 20"
                width="20"
              >
                <path
                  d="M14.348 14.849c-0.469 0.469-1.229 0.469-1.697 0l-2.651-3.030-2.651 3.029c-0.469 0.469-1.229 0.469-1.697 0-0.469-0.469-0.469-1.229 0-1.697l2.758-3.15-2.759-3.152c-0.469-0.469-0.469-1.228 0-1.697s1.228-0.469 1.697 0l2.652 3.031 2.651-3.031c0.469-0.469 1.228-0.469 1.697 0s0.469 1.229 0 1.697l-2.758 3.152 2.758 3.15c0.469 0.469 0.469 1.229 0 1.698z"
                />
              </svg>
            </div>
            <span
              class=" css-1xjgjl1-IndicatorSeparator"
            />
            <div
              aria-hidden="true"
              class=" css-1xc3v61-indicatorContainer"
            >
              <svg
                aria-hidden="true"
                class="css-tj5bde-Svg"
                focusable="false"
                height="20"
                viewBox="0 0 20 20"
                width="20"
              >
                <path
                  d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
      <div
        class="ascending tooltip"
        id="sort-direction"
      >
        <img
          alt="Ascending order"
          src="/assets/up_arrow.svg"
        />
      </div>
    </div>
  </form>
</div>
`;
